utcp.c

```diff
@@ -1042,6 +1053,7 @@ ssize_t utcp_recv(struct utcp *utcp, const void *data, size_t len) {
 
 	// 3. Check incoming data for acceptable seqno and update connection timer
 
+	size_t datalen = len;
```

Comment:
@a-krebs maybe instead we should introduce a variable packet_datalen and rename hdr packet_hdr so the len can be kept relatated to the data pointer: https://github.com/meshlink/utcp/blob/717092a2ece3c3a514472a6fe0683c6cfcba2e4d/utcp.c#L839